Fix Darwin compatibility

jvirtanen opened the pull request:
- Remove `ncurses` from checked modules as CMake cannot find `ncurses` on Darwin
- Add a compatibility header for endian conversions

jvirtanen committed:
Remove ncurses from checked modules

On Darwin, CMake cannot find ncurses.

Signed-off-by: Jussi Virtanen <jussi.k.virtanen@gmail.com>

jvirtanen committed:
Add compatibility header for endian conversions

Signed-off-by: Jussi Virtanen <jussi.k.virtanen@gmail.com>

jvirtanen commented:
I would be happy to find an alternative solution to the `ncurses` issue.

penberg commented on CMakeLists.txt:
```diff
@@ -18,7 +18,7 @@ find_package(Boost REQUIRED)
 find_package(PkgConfig)
 
-pkg_check_modules(LIBUV REQUIRED libuv>=1.0 ncurses)
+pkg_check_modules(LIBUV REQUIRED libuv>=1.0)
```
@jvirtanen, does this work on OS X:

``` diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08f9b3a..7e41a01 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,9 +16,11 @@ endif(DOXYGEN_FOUND)

 find_package(Boost REQUIRED)

+find_package(Curses)
+
 find_package(PkgConfig)

-pkg_check_modules(LIBUV REQUIRED libuv>=1.0 ncurses)
+pkg_check_modules(LIBUV REQUIRED libuv>=1.0)
 include_directories(${LIBUV_INCLUDE_DIRS})

 include_directories("include")
```

jvirtanen committed:
Find package Curses

This works on Darwin as well, unlike having ncurses in checked modules.

Suggested-by: Pekka Enberg <penberg@iki.fi>
Signed-off-by: Jussi Virtanen <jussi.k.virtanen@gmail.com>
